File: brasilburger-java/src/main/java/sn/ism/brasilburger/model/Menu.java
package sn.ism.brasilburger.model;

import java.math.BigDecimal;

public class Menu {
    private int id;
    private String nom;
    private String image;
    private BigDecimal prix;
    private boolean archived;

    public Menu() {
    }

    public Menu(int id, String nom, String image, BigDecimal prix, boolean archived) {
        this.id = id;
        this.nom = nom;
        this.image = image;
        this.prix = prix;
        this.archived = archived;
    }

    public Menu(String nom, String image, BigDecimal prix) {
        this(0, nom, image, prix, false);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public BigDecimal getPrix() {
        return prix;
    }

    public void setPrix(BigDecimal prix) {
        this.prix = prix;
    }

    public boolean isArchived() {
        return archived;
    }

    public void setArchived(boolean archived) {
        this.archived = archived;
    }

    @Override
    public String toString() {
        return "Menu{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", image='" + image + '\'' +
                ", prix=" + prix +
                ", archived=" + archived +
                '}';
    }
}

File: src/Entity/Menu.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'menu')]
class Menu
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_menu', type: 'integer')]
    private ?int $idMenu = null;

    #[ORM\Column(type: 'string', length: 120)]
    private string $nom;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private string $prix;

    #[ORM\Column(type: 'boolean', options: ['default' => false])]
    private bool $isArchived = false;

    #[ORM\ManyToMany(targetEntity: Burger::class)]
    #[ORM\JoinTable(name: 'menu_burger')]
    #[ORM\JoinColumn(name: 'id_menu', referencedColumnName: 'id_menu')]
    #[ORM\InverseJoinColumn(name: 'id_burger', referencedColumnName: 'id_burger')]
    private Collection $burgers;

    #[ORM\ManyToMany(targetEntity: Complement::class)]
    #[ORM\JoinTable(name: 'menu_complement')]
    #[ORM\JoinColumn(name: 'id_menu', referencedColumnName: 'id_menu')]
    #[ORM\InverseJoinColumn(name: 'id_complement', referencedColumnName: 'id_complement')]
    private Collection $complements;

    public function __construct()
    {
        $this->burgers = new ArrayCollection();
        $this->complements = new ArrayCollection();
    }

    public function getIdMenu(): ?int { return $this->idMenu; }
    public function getNom(): string { return $this->nom; }
    public function setNom(string $nom): self { $this->nom = $nom; return $this; }
    public function getImage(): ?string { return $this->image; }
    public function setImage(?string $image): self { $this->image = $image; return $this; }
    public function getPrix(): string { return $this->prix; }
    public function setPrix(string $prix): self { $this->prix = $prix; return $this; }
    public function isArchived(): bool { return $this->isArchived; }
    public function setIsArchived(bool $v): self { $this->isArchived = $v; return $this; }

    /** @return Collection<int, Burger> */
    public function getBurgers(): Collection { return $this->burgers; }

    /** @return Collection<int, Complement> */
    public function getComplements(): Collection { return $this->complements; }
}
